Source/Scene/Vector3DTileContent.js

```diff
@@ -0,0 +1,324 @@
+        byteOffset += sizeOfUint32;  // Skip magic number
+
+        //>>includeStart('debug', pragmas.debug);
+        var version = view.getUint32(byteOffset, true);
+        if (version !== 1) {
+            throw new DeveloperError('Only Vector tile version 1 is supported.  Version ' + version + ' is not.');
+        }
+        //>>includeEnd('debug');
+        byteOffset += sizeOfUint32;
+
+        var byteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
+        if (byteLength === 0) {
+            this.state = Cesium3DTileContentState.PROCESSING;
+            this.contentReadyToProcessPromise.resolve(this);
+            this.state = Cesium3DTileContentState.READY;
+            this.readyPromise.resolve(this);
+            return;
+        }
+
+        var featureTableJSONByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var featureTableBinaryByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var batchTableJSONByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var batchTableBinaryByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var indicesByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var positionByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
+        var featureTableString = getStringFromTypedArray(uint8Array, byteOffset, featureTableJSONByteLength);
+        var featureTableJSON = JSON.parse(featureTableString);
+        byteOffset += featureTableJSONByteLength;
+
+        var featureTableBinary = new Uint8Array(arrayBuffer, byteOffset, featureTableBinaryByteLength);
+        byteOffset += featureTableBinaryByteLength;
+
+        var numberOfPolygons = featureTableJSON.NUMBER_OF_POLYGONS;
+
+        var batchTableResources = new Cesium3DTileBatchTableResources(this, numberOfPolygons, createColorChangedCallback(this));
+        this.batchTableResources = batchTableResources;
+        if (batchTableJSONByteLength > 0) {
+            var batchTableString = getStringFromTypedArray(uint8Array, byteOffset, batchTableJSONByteLength);
+            batchTableResources.batchTable = JSON.parse(batchTableString);
+            byteOffset += batchTableJSONByteLength;
+        }
+
+        // TODO: Right now batchTableResources doesn't support binary
+        byteOffset += batchTableBinaryByteLength;
+
+        var indices = new Uint32Array(arrayBuffer, byteOffset, indicesByteLength / sizeOfUint32);
+        byteOffset += indicesByteLength;
+        var positions = new Uint16Array(arrayBuffer, byteOffset, positionByteLength / sizeOfUint16);
+
+        byteOffset = featureTableBinary.byteOffset + featureTableJSON.POLYGON_POSITION_OFFSETS.offset;
+        var offsets = new Uint32Array(featureTableBinary.buffer, byteOffset, numberOfPolygons);
```

Comment:
You may find it useful to use the `Cesium3DTileFeatureTableResources` that's in https://github.com/AnalyticalGraphicsInc/cesium/pull/4101.